templates/goshared/repeated.go

```diff
@@ -3,6 +3,10 @@ package goshared
 const repTpl = `
 	{{ $f := .Field }}{{ $r := .Rules }}
 
+	{{if .Rules.GetOmitempty}}
```

Comment:
I don't think you can rely on this for repeated fields, I think you need to do `len(field) != 0`. Adding `omitempty` for repeated fields is a bit weird as it is, but I can see it making sense (either empty, or minimum length 10, for example).

Reply:
This is exactly why I added this.

Reply:
Heh well there you go :-) I still would be worried about checking against nil vs len == 0 however, I would check against len == 0 to be safe.

Reply:
>  I would check against len == 0 to be safe.

Definitely agree with this, applicable to `map`, `repeated`, and `bytes` fields. While Go differentiates between nil and empty versions of these types, proto3 does not.

Reply:
Ok